src/main/java/edu/kis/powp/jobs2d/command/CommandCopier.java

```diff
@@ -0,0 +1,27 @@
+package edu.kis.powp.jobs2d.command;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class CommandCopier {
+	public static DriverCommand copyCommand(DriverCommand toCopy) {
+
+		DriverCommand copy = null;
+
+		try {
+			copy = toCopy.clone();
+			if (toCopy instanceof ICompoundCommand) {
+				Iterator<DriverCommand> iter = ((ICompoundCommand) toCopy).iterator();
+				List<DriverCommand> commandList = new ArrayList<>();
+				while (iter.hasNext()) {
+					commandList.add(iter.next().clone());
```

Comment:
I wonder: what would be the difference between calling clone() and copyCommand (recursively). I think that default implementation of clone can be a part of ICompoundCommand interface (using iterator operation and clone).